### ISSUE TITLE
MeshCentral Router + cockpit times out after 5 minutes

### ISSUE THREAD

**Issue by edlins** [Ylianst/MeshCentral#1950]
Hello,

I am running `cockpit` 215-1 (https://packages.ubuntu.com/bionic-backports/cockpit) to manage and monitor the OS on an Ubuntu Desktop 18.04.5 LTS (GDM3 on X11, not Wayland).  I can connect to the Ubuntu device in a MeshCentral 0.5.37 Desktop connection, run a `chromium-browser` on the device, navigate to ` cockpit` at https://localhost:9090, and login to `cockpit`, but that uses a lot of CPU and network bandwidth.  What I am trying to do is run Brave browser locally and forward port 9090 to the Ubuntu device via MeshCentral Router.  I do this by running:
```
MeshCentralRouter.exe -debug -host:[MeshCentral 0.5.37 server] -user:[xxx] -pass:[xxx] -map:TCP:9090:"[Ubuntu device name]"::9090
```
Then I can navigate to https://localhost:9090 from my local desktop to view the `cockpit` app remotely.

This works and saves a ton of CPU and bandwidth but for some reason it always times out after exactly 5 minutes regardless of user activity on the `cockpit` pages.  The `cockpit` page goes to `Disconnected` `Server has closed the connection.`  If I let it sit for 15 minutes it then switches to `You have been logged out due to inactivity`.  That's the `cockpit` default 15-minute inactivity timeout.  I can reproduce the 5-minute timeout with other web apps such as `portainer` so I'm fairly certain it's not a `cockpit` config issue.

**[EDIT: this error may be unrelated, see subsequent comment below]** I've looked through the `debug.log` and the only thing that jumps out at me is this message, where the [public IP addr] is the MeshCentral server:
```
[...]
Debug-Websocket got binary data, len = 1531
Debug-MeshMapper-#13: Websocket binary data: 1531
Debug-Websocket TCP failed to connect: System.Net.Sockets.SocketException (0x80004005): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond [public IP addr]:443
   at System.Net.Sockets.Socket.InternalEndConnect(IAsyncResult asyncResult)
   at System.Net.Sockets.Socket.EndConnect(IAsyncResult asyncResult)
   at System.Net.Sockets.TcpClient.EndConnect(IAsyncResult asyncResult)
   at MeshCentralRouter.webSocketClient.OnConnectSink(IAsyncResult ar)
Debug-Websocket got binary data, len = 4096
Debug-MeshMapper-#13: Websocket binary data: 4096
[...]
```
If I can get this stable, `cockpit` over MeshCentral Router would be a very nice combo for remotely managing linux devices (or running any webapps remotely).  I believe this is a hugely valuable use case.  I can provide the `debug.log` but I'd rather not post it here as it has email addresses, server hostnames, etc.  This should be easy enough to reproduce, and it would be nice to know it's not just me.

Thanks for the great service, and any help you can provide!

**Comment by edlins** [comment]
I just ran this again and monitored in real-time and it appears that this is actually what happens at the time of the disconnect:
```
Debug-Websocket got binary data, len = 101
Debug-MeshMapper-#11: Websocket binary data: 101
Debug-Websocket got string data, len = 38
Debug-MeshMapper-#11: Websocket got text frame: {"ctrlChannel":"102938","type":"pong"}
Debug-Websocket got closed fragment.
Debug-MeshMapper-#11: Websocket mapping, connected to server.
Debug-MeshMapper-#11: ShutdownWebClients()
Debug-MeshMapper-#11: ShutdownWebClients()
```

**Comment by edlins** [comment]
And I saw this on the Ubuntu device:
```
Nov  4 14:20:26 Seneca-PRO794675-1978164 cockpit-ws[23994]: received fragmented control frame
Nov  4 14:20:26 Seneca-PRO794675-1978164 cockpit-ws[23994]: Received invalid WebSocket response from the server
Nov  4 14:20:26 Seneca-PRO794675-1978164 cockpit-ws[23994]: Connection to session 1451 closed
Nov  4 14:20:42 Seneca-PRO794675-1978164 cockpit-ws[23994]: session timed out
```

**Comment by edlins** [comment]
Tried editing `nginx` config with the following, no change:
```
    # MeshCentral uses long standing web socket connections, set longer timeouts.
    proxy_send_timeout 28800s;
    proxy_read_timeout 28800s;
```
Still timed out after 5 minutes.  Might try adjusting MC `config.json` next.  Any suggestions welcome.

**Comment by edlins** [comment]
I have a finding on this.  I edited the `config.json` to change the `AgentPing` and `AgentPong` settings.  When I set either `AgentPing` or `AgentPong` to 10s, the connection terminated after 10s.  When I removed both settings entirely, the connection no longer terminates!  This can be reproduced easily by setting the following in `config.json`
```
"AgentPing": 10,
"AgentPong": 10,
```
then connect to a linux agent and run
```
sudo apt-get install -y cockpit
```
Then setup routing like above with
```
MeshCentralRouter.exe -debug -host:[MeshCentral 0.5.37 server] -user:[xxx] -pass:[xxx] -map:TCP:9090:"[Ubuntu device name]"::9090
```
Then run a browser locally and connect to
```
https://localhost:9090
```
and login.  The connection should time out after 10s.  Remove both of those settings from `config.json` and the connection will run until the cockpit default timeout of 15m is hit.
